torch/testing/_internal/common_modules.py

```diff
@@ -2524,6 +2526,8 @@ def module_error_inputs_torch_nn_Pad3d(module_info, device, dtype, requires_grad
                ),
     ModuleInfo(torch.nn.AdaptiveAvgPool2d,
                gradcheck_nondet_tol=GRADCHECK_NONDET_TOL,
+               # Fails on backward check if output size is 1x1
+               gradcheck_memformat=False,
```

Comment:
Really appreciate the detailed comments here regarding why each of these is set to False! 
 
nit: Separately, is there any way we could make this an xfail? I'm hoping that if these are fixed, whoever sends the PR fixing will get signal to un-xfail these tests. I worry that if we make this a flag, re-enabling the tests might fall through the cracks.

I'm thinking maybe something to the effect of 
```
DecorateInfo(
     unittest.expectedFailure, 'TestModule',
     'test_memory_format',
     active_if=lambda p: p['training']
 )
```
This seems reasonable to me because the non-backward version will still be tested if training=False and we can change the check [here](https://github.com/pytorch/pytorch/blob/499aaf6cd790ba1008fbacabf8b0fd26d93ca289/test/test_modules.py#L693) to `if training and len(ref_diff_outputs) > 0`

Let me know your thoughts on this!

Reply:
Thanks for your suggestion! It‘s a good idea. I just found there is `active_if` by which `DecorateInfo` can be activated according to input parameters of the test.
Modified as suggested.